cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -639,12 +639,8 @@ class InputState : public util::SerialSequencingQueue::Processor {
         // hit the end of the batch, need to get the next batch if possible.
         ++batches_processed_;
         latest_ref_row_ = 0;
-        have_active_batch &= !queue_.TryPop();
-        if (have_active_batch) {
-          DCHECK_GT(queue_.Front()->num_rows(), 0);  // empty batches disallowed
-          memo_.UpdateTime(GetTime(queue_.Front().get(), time_type_id_, time_col_index_,
-                                   0));  // time changed
-        }
+        std::ignore = queue_.TryPop();
```

Comment:
Perhaps we can add some debug checking?
```suggestion
        bool did_pop = queue_.TryPop().has_value();
        DCHECK(did_pop);
        ARROW_UNUSED(did_pop);
```

Reply:
This is definitely a good point, will do. Thank you!